dist/nix/vimrc-with-plugins.nix

```diff
@@ -0,0 +1,28 @@
+{ pkgs ? import <nixpkgs>{}
+, additionalPlugins? []
+, additionalCustPlugins? {}
+, additionalVimrc? ""#(pkgs.builtins.readFile ./vimrc)
+}:
+with pkgs;
+let quick-lint-js = import ./quick-lint-js.nix { inherit pkgs;} ;
+in
+vimUtils.vimrcFile {
+  customRC =
+    ''
+      let g:ale_javascript_quick_lint_js_executable = '${quick-lint-js}/bin/quick-lint-js'
+    ''+additionalVimrc;
```

Comment:
Instead of inserting a line into the vimrc, we could generate a `plugin/quick-lint-js-nix.vim` file in the `buildPhase` of the plugin derivation (line 15).